### ISSUE TITLE
inserting data into mongodb fails with teleport 11

### ISSUE THREAD

**Issue by blushingpenguin** [gravitational/teleport#18549]
Expected behavior:
Inserting data succeeds

Current behavior:
```
rs0:PRIMARY> db.getCollection('Test').insert({foo:"bar"})
Error: Mismatched RPC protocols - request was 'msg' ' but reply was 'reply'  :
runClientFunctionWithRetries@src/mongo/shell/session.js:361:27
runCommand@src/mongo/shell/session.js:455:25
DB.prototype._runCommandImpl@src/mongo/shell/db.js:147:12
DB.prototype.runCommand@src/mongo/shell/db.js:162:16
DBCollection.prototype._dbCommand@src/mongo/shell/collection.js:176:16
executeBatch@src/mongo/shell/bulk_api.js:912:22
Bulk/this.execute@src/mongo/shell/bulk_api.js:1163:21
DBCollection.prototype.insert@src/mongo/shell/collection.js:322:22
```
which is being run from a client spawned by `tsh db connect --db-name=XXX YYY`. The server is deployed by the teleport-cluster helm chart, and the connection is routed via a proxy deployed by the teleport-kube-agent chart. 

The teleport client, server and kube agent version are all 11.0.3 (Teleport v11.0.3 git:v11.0.3-0-g6f0eeef go1.19.2). This used to work before upgrading from 10.3.2, and if I downgrade teleport-kube-agent to 10.3.8 (leaving the other components at 11.0.3) then inserts start working again.

Turning debug logs on for the proxy and running the insert reports:

```
2022-11-17T06:22:15Z INFO [AUDIT]     db.session.end cluster_name:teleport.example.com code:TDB01I db_name:Integrations db_protocol:mongodb db_service:dev-mongo db_uri:mongodb://mongo-0.mongo.dev.svc.cluster.local:27017/?replicaSet=rs0 db_user:mark.weaver ei:12 event:db.session.end sid:8edc50bb-0add-4834-997e-88f1dd0131ca time:2022-11-17T06:22:15.082Z uid:8e843fec-2aad-445b-ba96-db042a59c565 user:mark.weaver events/emitter.go:263
2022-11-17T06:22:15Z ERRO [DB:SERVIC] Failed to handle connection. addr:10.152.183.252:443 error:[
ERROR REPORT:
Original Error: *trace.BadParameterError malformed OP_MSG: invalid document length [0 0 0 0 1 49 0 0 0 100 111 99 117 109 101 110 116 115 0 35 0 0 0 7 95 105 100 0 99 117 211 22 180 238 168 111 116 12 133 138 2 102 111 111 0 4 0 0 0 98 97 114 0 0 0 192 0 0 0 2 105 110 115 101 114 116 0 5 0 0 0 116 101 115 116 0 8 111 114 100 101 114 101 100 0 1 3 108 115 105 100 0 30 0 0 0 5 105 100 0 16 0 0 0 4 191 130 183 99 242 214 79 62 144 79 249 110 103 181 70 59 0 3 36 99 108 117 115 116 101 114 84 105 109 101 0 88 0 0 0 17 99 108 117 115 116 101 114 84 105 109 101 0 1 0 0 0 15 211 117 99 3 115 105 103 110 97 116 117 114 101 0 51 0 0 0 5 104 97 115 104 0 20 0 0 0 0 245 222 197 251 29 164 88 211 234 96 234 210 25 178 187 15 26 74 42 174 18 107 101 121 73 100 0 6 0 0 0 81 216 198 98 0 0 2 36 100 98 0 13 0 0 0 73 110 116 101 103 114 97 116 105 111 110 115 0 0]
Stack Trace:
	github.com/gravitational/teleport/lib/srv/db/mongodb/protocol/opmsg.go:256 github.com/gravitational/teleport/lib/srv/db/mongodb/protocol.readOpMsg
	github.com/gravitational/teleport/lib/srv/db/mongodb/protocol/message.go:58 github.com/gravitational/teleport/lib/srv/db/mongodb/protocol.ReadMessage
	github.com/gravitational/teleport/lib/srv/db/mongodb/engine.go:92 github.com/gravitational/teleport/lib/srv/db/mongodb.(*Engine).HandleConnection
	github.com/gravitational/teleport/lib/srv/db/server.go:812 github.com/gravitational/teleport/lib/srv/db.(*Server).handleConnection
	github.com/gravitational/teleport/lib/srv/db/server.go:722 github.com/gravitational/teleport/lib/srv/db.(*Server).HandleConnection
	github.com/gravitational/teleport/lib/reversetunnel/transport.go:277 github.com/gravitational/teleport/lib/reversetunnel.(*transport).start
	github.com/gravitational/teleport/lib/reversetunnel/agent.go:581 github.com/gravitational/teleport/lib/reversetunnel.(*agent).handleDrainChannels.func2
	runtime/asm_amd64.s:1594 runtime.goexit
User Message: malformed OP_MSG: invalid document length [0 0 0 0 1 49 0 0 0 100 111 99 117 109 101 110 116 115 0 35 0 0 0 7 95 105 100 0 99 117 211 22 180 238 168 111 116 12 133 138 2 102 111 111 0 4 0 0 0 98 97 114 0 0 0 192 0 0 0 2 105 110 115 101 114 116 0 5 0 0 0 116 101 115 116 0 8 111 114 100 101 114 101 100 0 1 3 108 115 105 100 0 30 0 0 0 5 105 100 0 16 0 0 0 4 191 130 183 99 242 214 79 62 144 79 249 110 103 181 70 59 0 3 36 99 108 117 115 116 101 114 84 105 109 101 0 88 0 0 0 17 99 108 117 115 116 101 114 84 105 109 101 0 1 0 0 0 15 211 117 99 3 115 105 103 110 97 116 117 114 101 0 51 0 0 0 5 104 97 115 104 0 20 0 0 0 0 245 222 197 251 29 164 88 211 234 96 234 210 25 178 187 15 26 74 42 174 18 107 101 121 73 100 0 6 0 0 0 81 216 198 98 0 0 2 36 100 98 0 13 0 0 0 73 110 116 101 103 114 97 116 105 111 110 115 0 0]] db/server.go:724
2022-11-17T06:22:15Z DEBU [PROXY:AGE] Transport request: teleport-transport. leaseID:40 target:teleport.example.com:443 cluster:teleport.example.com reversetunnel/agent.go:570
2022-11-17T06:22:15Z DEBU [PROXY:AGE] Received out-of-band proxy transport request for @local-node [86064deb-1390-41e2-85aa-1097ab22984f.teleport.example.com]. cluster:teleport.example.com reversetunnel/transport.go:201
2022-11-17T06:22:15Z DEBU [PROXY:AGE] Handing off connection to a local "db" service. cluster:teleport.example.com reversetunnel/transport.go:276
```

**Comment by gabrielcorado** [comment]
@blushingpenguin With my standard setup, I wasn't able to reproduce it. Can you provide the following information so I can form an environment closer to what you have:
* Which version is your MongoDB? And which version is your Mongosh?
* Are you using MongoDB sharding?
* Is your Teleport Proxy using TLS routing? `auth_service.proxy_listener_mode: multiplex` or `proxyListenerMode: "multiplex"` Helm value

Thanks!

**Comment by blushingpenguin** [comment]
@gabrielcorado 
* mongo 4.4.15 installed from running on ubuntu 20.04 in a kubernetes container (https://www.mongodb.com/docs/v4.4/tutorial/install-mongodb-on-ubuntu/)
* we aren't using sharding
* yes, the slightly redacted configs (both generated by the helm charts) are below

the main teleport config is:
```
version: v2
teleport:
  log:
    severity: INFO
    output: stderr
    format:
      output: text
      extra_fields: [timestamp, level, component, caller]
auth_service:
  enabled: true
  cluster_name: teleport.example.com
  proxy_listener_mode: multiplex
proxy_service:
  enabled: true
  public_addr: teleport.example.com:443
  https_keypairs:
  - key_file: /etc/teleport-tls/tls.key
    cert_file: /etc/teleport-tls/tls.crt
ssh_service:
  enabled: false
```

the slightly redacted proxy config  is:
```
teleport:
  join_params:
    method: "token"
    token_name: "/etc/teleport-secrets/auth-token"
  auth_servers: ["teleport.examplecom:443"]
  log:
    severity: DEBUG
    output: stderr
    format:
      output: text
      extra_fields: ["timestamp","level","component","caller"]

db_service:
  enabled: true
  databases:
  - description: dev mongo
    name: dev-mongo
    protocol: mongodb
    static_labels:
      db: mongo
      deployment: dev
    tls:
      ca_cert_file: /db-cas/dev-ca.crt
      mode: verify-full
    uri: mongodb://mongo-0.mongo.dev.svc.cluster.local:27017/?replicaSet=rs0
```

**Comment by smallinsky** [comment]
Fixed in Teleport 11.1.0 by https://github.com/gravitational/teleport/pull/18690
